amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/catalog/FlinkUnifiedCatalog.java

```diff
@@ -221,8 +226,25 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     TableIdentifier tableIdentifier =
         TableIdentifier.of(
             unifiedCatalog.name(), tablePath.getDatabaseName(), tablePath.getObjectName());
+    if (!configuration.contains(TABLE_FORMAT)) {
+      // if user doesn't specify the table format, we try to load the table and get the table
+      // format.
+      try {
+        AmoroTable amoroTable =
+            unifiedCatalog.loadTable(tableIdentifier.getDatabase(), tableIdentifier.getTableName());
+        format = amoroTable.format();
+      } catch (Throwable t) {
+        LOG.warn(
+            "We can't load table {}.{} through unifiedCatalog, use default format {}",
+            tablePath.getDatabaseName(),
+            tablePath.getObjectName(),
+            format.name());
+      }
+    }
+    final TableFormat catalogFormat = format;
     AbstractCatalog catalog =
-        getOriginalCatalog(format).orElseGet(() -> createOriginalCatalog(tableIdentifier, format));
+        getOriginalCatalog(format)
+            .orElseGet(() -> createOriginalCatalog(tableIdentifier, catalogFormat));
```

Comment:
If table exists, we load table and get tableFormat for it, Otherwise, we use default tableFormat to create table, which is MIXED_ICEBERG.

Did we should force user to add 'table.format' option in create table statement?

Reply:
> Did we should force user to add 'table.format' option in create table statement?
Good idea. If the table cannot be loaded via unifiedCatalog and no format is specified, we should report an error to prompt the user to specify the format. WDYT?

Reply:
I think we could support a property of catalog which define a defult table format when creating a table without specific the format.

Reply:
> I think we could support a property of catalog which define a defult table format when creating a table without specific the format.

IMO, Configuring the format on both the table and the catalog at the same time can easily mislead users. The "Unified Catalog" is designed to allow users not to worry about the underlying format (except when it is created).

Let's listen to some other opinions.   WDYT  @YesOrNo828  @xieyi888